test/libsolidity/SolidityExecutionFramework.cpp

```diff
@@ -28,12 +29,27 @@ using namespace dev;
 using namespace dev::solidity;
 using namespace dev::solidity::test;
 
+ExecutionFramework::boostArg::boostArg()
+{
+	ipcPath = "~/.ethereum/geth.ipc";
+	size_t argc = boost::unit_test::framework::master_test_suite().argc;
+	char** argv = boost::unit_test::framework::master_test_suite().argv;
+	for (size_t i = 0; i < argc; i++)
+	{
+		string arg = argv[i];
+		if (arg == "--ipc" && i+1 < argc)
+		{
+			ipcPath = argv[i+1];
+			i++;
+		}
+	}
+}
 
-ExecutionFramework::ExecutionFramework():
-	m_rpc(RPCSession::instance("/home/wins/Ethereum/testnet/ethnode1/geth.ipc")),
+ExecutionFramework::ExecutionFramework() :
+	m_boostArg(),
+	m_rpc(RPCSession::instance(m_boostArg.ipcPath)),
```

Comment:
Why does it have to be a member? Wouldn't a simple free function `parseIPCPathFromArguments` that returns a string and calls BOOST_FAIL otherwise also work?